.github/workflows/snapshot-releases.yaml

```diff
@@ -0,0 +1,34 @@
+name: Snapshot Releases
+
+
+# Only run on pushes to main:
+# Replace with 'on: [push]' if you need to run tests on your PR
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          # We could consider bumping this for snapshots?
+          java-version: '8'
+          distribution: 'temurin'
+          server-id: ossrh
+          server-username: OSSRH_USERNAME
+          server-password: OSSRH_PASSWORD
+      - name: Publish package
+        env:
+          OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+          OSSRH_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
+          # These don't work with java 8.
+          # Consider have something separate with more recent versions for the modules that don't support older versions?
```

Comment:
might use matrix approach as we do in https://github.com/langchain4j/langchain4j/blob/main/.github/workflows/main.yaml

Reply:
Indeed that would make sense. 

What is the approach for releases? Is everything released from the older version of Java that is possible?

If you have a script or a bunch of options that are used for release they could be useful here, otherwise I can cobble something together from the existing workflow files.

Reply:
> What is the approach for releases? Is everything released from the older version of Java that is possible?

Yes,
- we release with java 8 everything except `langchain4j-opensearch`, `langchain4j-neo4j` and `langchain4j-code-execution-engine-graalvm-polyglot`
- then `langchain4j-opensearch` with 11
- then `langchain4j-neo4j` and `langchain4j-code-execution-engine-graalvm-polyglot` with 17

I do not have any script unfortunately.

Reply:
I'll have a look, I think that should be feasible in some way

Reply:
One side question though: is it important that all snapshots for a release from master share the same version?

Currently it seems based on time: I assume we could have something based on the commit hash if required.

Reply:
Yes, I guess users should be able to use `X.Y.Z-SNAPSHOT` version and maven should provide the latest snapshot, or?